### PR TITLE
test(test_basic.py): add OCR and encrypted PDF filter tests

### PR DESCRIPTION
**Pull request by nicholaschiang**
Also fixes some bugs in the `filter_encrypted()` function found in core.py:

This will change the regex pattern to search bytes instead of strings as `r = requests.get("https://example.com")` returns a byte-like object when written as `target_file.write(r.content)`, not what was previously assumed when using urllib2 in `download_pdf()`.

### CODE REVIEW

**Comment by nicholaschiang** [comment]
Also adds my significantly smaller `tests/data/` which includes some basic CSVs and PDFs for use by `test_basic.py`.

File: tests/data/download-csv/csv/ffe002af_0.csv [hunk 1]
```diff
@@ -0,0 +1 @@
...px 0 2px rgba(0,0,0,0.3) inset,      0 1px 2px rgba(0,0,0,0.2) inset,      0 -1px 2px rgba(255,255,255,0.3) inset,      1px 2px 2px rgba(0,0,0,0.2) inset;  }  .circle_inner {    box-shadow:      1px 0 2px rgba(0,0,0,0.3),      0 1px 2px rgba(0,0,0,0.2),      0 -1px 2px rgba(255,255,255,0.3),      1px 2px 2px rgba(0,0,0,0.2);  }  .circle_inner .center {    position: absolute;    top: 75px;    left: 0px;    right: 0px;    text-align: center;  }  h2 {    margin: 15px 0px;    font-weight: 500;    font-size: 1.4em;    line-height: 1.3em;  }  .actions {    margin: 30px 0px;    font-size: 1.3em;    line-height: 1.3em;    font-weight: 600;  }  .details {    text-align: right;    margin: 100px 20px 0px;    font-size: 0.7em;    line-height: 1.3em;  }</style></head><body><p><br><p style="font-size:medium"><center>Web request id: 7902291853812073919:FWP01-CPVNPPRY11:1547346302:0</center><p><br><p style="font-size:medium"><center><a href="mailto:blocked@digikey.com?Subject=7902291853812073919:FWP01-CPVNPPRY11:1547346302:0">Report this error</a><br><br><br></center></p></p><div class='circle_outer'><div class='circle_inner'><div class='center'>    <h1>403</h1>  </div></div></div>  </body></html>
```
Comment (lukehsiao) [inline tests/data/download-csv/csv/ffe002af_0.csv]: This isn't a CSV...

File: tests/data/download-csv/csv/ffe002af_1.csv [hunk 2]
```diff
@@ -0,0 +1 @@
...px 0 2px rgba(0,0,0,0.3) inset,      0 1px 2px rgba(0,0,0,0.2) inset,      0 -1px 2px rgba(255,255,255,0.3) inset,      1px 2px 2px rgba(0,0,0,0.2) inset;  }  .circle_inner {    box-shadow:      1px 0 2px rgba(0,0,0,0.3),      0 1px 2px rgba(0,0,0,0.2),      0 -1px 2px rgba(255,255,255,0.3),      1px 2px 2px rgba(0,0,0,0.2);  }  .circle_inner .center {    position: absolute;    top: 75px;    left: 0px;    right: 0px;    text-align: center;  }  h2 {    margin: 15px 0px;    font-weight: 500;    font-size: 1.4em;    line-height: 1.3em;  }  .actions {    margin: 30px 0px;    font-size: 1.3em;    line-height: 1.3em;    font-weight: 600;  }  .details {    text-align: right;    margin: 100px 20px 0px;    font-size: 0.7em;    line-height: 1.3em;  }</style></head><body><p><br><p style="font-size:medium"><center>Web request id: 1437497647068257654:FWP01-CPVNPPRY17:1547346302:0</center><p><br><p style="font-size:medium"><center><a href="mailto:blocked@digikey.com?Subject=1437497647068257654:FWP01-CPVNPPRY17:1547346302:0">Report this error</a><br><br><br></center></p></p><div class='circle_outer'><div class='circle_inner'><div class='center'>    <h1>403</h1>  </div></div></div>  </body></html>
```
Comment (lukehsiao) [inline tests/data/download-csv/csv/ffe002af_1.csv]: this isn't CSV either....

File: tests/data/download-pdf/csv/ffe002af_0.csv [hunk 3]
```diff
@@ -0,0 +1,2 @@
+"ï»¿Datasheets,Image,Digi-Key Part Number",Manufacturer Part Number,Manufacturer,Description,Quantity Available,Factory Stock,Unit Price (USD),@ qty,Minimum Quantity,Packaging,Series,Part Status,Amplifier Type,Number of Circuits,Output Type,Slew Rate,Gain Bandwidth Product,-3db Bandwidth,Current - Input Bias,Voltage - Input Offset,Current - Supply,Current - Output / Channel,"Voltage - Supply, Single/Dual (Â±)",Operating Temperature,Mounting Type,Package / Case,Supplier Device Package
```
Comment (lukehsiao) [inline tests/data/download-pdf/csv/ffe002af_0.csv]: This is a CSV! But, why does it have weird chars at the start?

File: tests/data/download-csv/csv/ffe002af_2.csv [hunk 4]
```diff
@@ -0,0 +1 @@
...px 0 2px rgba(0,0,0,0.3) inset,      0 1px 2px rgba(0,0,0,0.2) inset,      0 -1px 2px rgba(255,255,255,0.3) inset,      1px 2px 2px rgba(0,0,0,0.2) inset;  }  .circle_inner {    box-shadow:      1px 0 2px rgba(0,0,0,0.3),      0 1px 2px rgba(0,0,0,0.2),      0 -1px 2px rgba(255,255,255,0.3),      1px 2px 2px rgba(0,0,0,0.2);  }  .circle_inner .center {    position: absolute;    top: 75px;    left: 0px;    right: 0px;    text-align: center;  }  h2 {    margin: 15px 0px;    font-weight: 500;    font-size: 1.4em;    line-height: 1.3em;  }  .actions {    margin: 30px 0px;    font-size: 1.3em;    line-height: 1.3em;    font-weight: 600;  }  .details {    text-align: right;    margin: 100px 20px 0px;    font-size: 0.7em;    line-height: 1.3em;  }</style></head><body><p><br><p style="font-size:medium"><center>Web request id: 6536369947070495016:FWP01-CPVNPPRY16:1547346302:0</center><p><br><p style="font-size:medium"><center><a href="mailto:blocked@digikey.com?Subject=6536369947070495016:FWP01-CPVNPPRY16:1547346302:0">Report this error</a><br><br><br></center></p></p><div class='circle_outer'><div class='circle_inner'><div class='center'>    <h1>403</h1>  </div></div></div>  </body></html>
```
Comment (lukehsiao) [inline tests/data/download-csv/csv/ffe002af_2.csv]: Another HTML file...

File: tests/test_basic.py [hunk 5]
```diff
@@ -115,18 +123,70 @@ def test_download_pdf():
...
+    r = requests.get(test_link)
+    outfile = test_dir + test_filename
+    with open(outfile, "wb") as target:
+        target.write(r.content)
+
+    # TODO: Add valid PDF to test_dir
+    # and test that filter_encrypted() keeps that file
+    # Check if output matches manully inputted test variables
+    if __debug__:
+        digikeyscraper.filter_encrypted(test_dir)
```
Comment (lukehsiao) [inline tests/test_basic.py]: Seems like these filter functions should be broken down to a function just returning a boolean of whether or not a file is encrypted and should be removed. Then the calling function should remove it. That way you can test it more directly.

File: digikeyscraper/core.py [hunk 6]
```diff
@@ -143,11 +160,17 @@ def download_pdf(csv_dir, pdf_dir):
     )
     total = len(final_unique_urls_and_filenames)
     total_count = 0
+    # Specify user agent to avoid error code 403
+    user_agent = (
+        "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36"
+        + "(KHTML, like Gecko) Chrome/70.0.3538.77 Safari/537.36"
```
Comment (lukehsiao) [inline digikeyscraper/core.py]: Seems inconvenient to have this defined in two separate places. If we want to change it, we only want to change it in one place.

File: digikeyscraper/core.py [hunk 7]
```diff
@@ -182,73 +205,78 @@ def download_pdf(csv_dir, pdf_dir):
...
+    # Use 'pdffonts' to check for PDFs that would require OCR
+    try:
+        output = subprocess.check_output(
+            ["pdffonts", pdf_filename], stderr=subprocess.STDOUT, encoding="utf-8"
+        )
+    except subprocess.CalledProcessError:
+        # Usually means we got an HTML file instead of an actual PDF.
+        return True
+    # output = subprocess.Popen(["pdffonts", pdf_filename],
+    # stdout=subprocess.PIPE).communicate()[0]
```
Comment (lukehsiao) [inline digikeyscraper/core.py]: clean up debugging comments

File: tests/test_basic.py [hunk 8]
```diff
@@ -115,18 +136,97 @@ def test_download_pdf():
...
+    with open(outfile, "wb") as target:
+        target.write(r.content)
+
+    # TODO: Add valid PDF to test_dir
+    # and test that filter_encrypted() keeps that file
+    # Check if output matches manully inputted test variables
+    if __debug__:
+        # Check if test file downloaded correctly
+        if not os.path.isfile(pdf_filename):
+            raise AssertionError
```
Comment (lukehsiao) [inline tests/test_basic.py]: You seem to like the `__debug__`. Just letting you know that this seems like it's simpler to just write as
```py
assert os.path.isfile(pdf_filename)
```
directly.

Same for most of the code you have in under a `__debug__`. Doing so is a little more clear and concise.

Comment (nicholaschiang) [reply tests/test_basic.py]: I changed most of those `__debug__` statements in my next commit, but I kept the overall `if __debug__:` to ensure that `test_basic.py` doesn't download CSVs without explicit permissions.

Comment (lukehsiao) [reply tests/test_basic.py]: > I kept the overall if __debug__: to ensure that test_basic.py doesn't download CSVs without explicit permissions.

What does this mean?

File: digikeyscraper/core.py (hunk 6 above)
Comment (nicholaschiang) [reply digikeyscraper/core.py]: Done. Defined `user_agent` once at the top of `core.py`.

**Pull request merged**